cpp/include/raft/core/resource/device_memory_resource.hpp

```diff
@@ -15,24 +15,49 @@
  */
 #pragma once
 
+#include <raft/core/operators.hpp>
 #include <raft/core/resource/resource_types.hpp>
 #include <raft/core/resources.hpp>
+#include <raft/util/cudart_utils.hpp>
+
 #include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/limiting_resource_adaptor.hpp>
 #include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+#include <cstddef>
+#include <optional>
 
 namespace raft::resource {
-class device_memory_resource : public resource {
+class limited_memory_resource : public resource {
```

Comment:
Small nitpick: the rmm memory resource is called a "limiting_memory_resource" but this class is called "limited_memory_resource". We should be consistent.

Reply:
I think it's probably best to keep the use of "limiting" here so that we are consistent with RMM's naming conventions as well.